core/io/ip.cpp

```diff
@@ -75,7 +75,7 @@ struct _IP_ResolverPrivate {
 	Semaphore sem;
 
 	Thread thread;
-	bool thread_abort = false;
+	SafeFlag thread_abort;
```

Comment:
The variable is synced with the semaphore. However, I see now that a data race is still possible, if `bool` doesn't provide atomicity guarantees across platforms. Therefore, thinking about this (and similar cases), my abandoned `RelaxedFlag` project is relevant (#63189). That one asks the implementation for atomicity, but without acquire-release semantics, which the semaphore is already providing.

Thoughts?

P.S.: What was the sanitizer complaint, by the way?

Reply:
Providing it is an IP hostname resolver, it's not performance critical and any overhead for `SafeFlag` going to be negligible compared to networking.